stl/inc/deque

```diff
@@ -1145,29 +1213,47 @@ public:
 #endif // _ITERATOR_DEBUG_LEVEL == 2
     }
 
-    template <class _Iter, enable_if_t<_Is_iterator_v<_Iter>, int> = 0>
-    void assign(_Iter _First, _Iter _Last) {
+private:
+    template <class _Iter, class _Sent>
+    void _Assign_range(_Iter _First, const _Sent _Last) {
         _Orphan_all();
-        _Adl_verify_range(_First, _Last);
-        auto _UFirst       = _Get_unwrapped(_First);
-        const auto _ULast  = _Get_unwrapped(_Last);
         auto _Myfirst      = _Unchecked_begin();
         const auto _Mylast = _Unchecked_end();
-        for (; _UFirst != _ULast; ++_UFirst) { // try to assign over an element in the container
-            if (_Myfirst == _Mylast) { // container wasn't big enough, insert what's left at end
-                do {
-                    emplace_back(*_UFirst);
-                    ++_UFirst;
-                } while (_UFirst != _ULast);
+        for (;;) { // try to assign over an element in the container

```

Comment:
Could we do something like:
```cpp
// try to assign over an element in the container
for (; _Myfirst != _Mylast; ++_Myfirst, (void) ++_First) {
    if (_First == _Last) {
        _Erase_last_n(static_cast<size_type>(_Mylast - _Myfirst));
        return;
    }

    *_Myfirst = *_First;
}

// container wasn't big enough, insert what's left at end
for (;_First != _Last; ++_First) {
    _Emplace_back_internal(*_First);
}
```

Reply:
Hell yes we can. Very nice simplification.